### ISSUE TITLE
useExternal: reference count for multiple instances unloading the same node

### ISSUE THREAD

**Issue by KrishnaPG** [alibaba/hooks#824]
````
    const [status, { unload }] = useExternal('/useExternal/stylesheet.css')
````

How to safely use multiple instances, all loading the same stylesheet, unload it cleanly?

For example, 4 instances of a component load the same stylesheet. When the components are about to unmount, we need to unload the injected css DOM node. But this removal should happen only when the last component unmounts. How to achieve it?

Somehow the hook needs to track the number of active references pending for it so that it can cleanly unload on the last item's exit.

**Comment by KrishnaPG** [comment]
Also, looks like the `useExternal` hook is not taking into consideration duplicate script/css nodes. It is preferable to add a check to find if script/css nodes already exist in the DOM before creating them, as done in this https://usehooks.com/useScript/

**Comment by brickspert** [comment]
@Turkyden can you support these features?

**Comment by turkyden** [comment]
OK, i wil check it.

**Comment by brickspert** [comment]
Hello：

这个问题我们已经在 v3.0 版本彻底修复，目前 v3.0 版本已经发布 alpha 版本。

欢迎试用。

文档：https://ahooks-next.surge.sh/
源码：https://github.com/alibaba/hooks/tree/next

安装最新分支
```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```

We have fixed this problem in the v3.0 version, and the alpha has been released. 

Welcome to try.

Doc：https://ahooks-next.surge.sh/
Source Code：https://github.com/alibaba/hooks/tree/next

Install v3.0

```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```
